kobuki_auto_docking/src/auto_docking_ros.cpp

```diff
@@ -12,159 +12,243 @@
-  as_.registerPreemptCallback(boost::bind(&AutoDockingROS::preemptCb, this));
-  as_.start();
-}
+    this->as_ = rclcpp_action::create_server<AutoDocking>(
+      this,
+      "auto_docking_action",
+      std::bind(&AutoDockingROS::handle_goal, this, _1, _2),
+      std::bind(&AutoDockingROS::handle_cancel, this, _1),
+      std::bind(&AutoDockingROS::handle_accepted, this, _1)
+    );
 
-AutoDockingROS::~AutoDockingROS()
-{
-  shutdown_requested_ = true;
-  if (as_.isActive()) {
-    result_.text = "Aborted: Shutdown requested.";
-    as_.setAborted( result_, result_.text );
-  }
-  dock_.disable();
-}
+    this->declare_parameter<double>("min_abs_v", 0.01);
+    this->declare_parameter<double>("min_abs_w", 0.1);
 
-bool AutoDockingROS::init(ros::NodeHandle& nh)
-{
-  // Configure docking drive
-  double min_abs_v, min_abs_w;
-  if (nh.getParam("min_abs_v", min_abs_v) == true)
-    dock_.setMinAbsV(min_abs_v);
+    // Configure docking drive
+    double min_abs_v, min_abs_w;
+    if (this->get_parameter("min_abs_v", min_abs_v) == true)
+      dock_.setMinAbsV(min_abs_v);
 
-  if (nh.getParam("min_abs_w", min_abs_w) == true)
-    dock_.setMinAbsW(min_abs_w);
+    if (this->get_parameter("min_abs_w", min_abs_w) == true)
+      dock_.setMinAbsW(min_abs_w);
 
-  // Publishers and subscribers
-  velocity_commander_ = nh.advertise<geometry_msgs::Twist>("velocity", 10);
-  debug_jabber_ = nh.advertise<std_msgs::String>("debug/feedback", 10);
+    // parater callback
+    param_callback_handle_ = this->add_on_set_parameters_callback(
+      std::bind(&AutoDockingROS::parametersCallback, this, std::placeholders::_1)
+    );
 
-  debug_ = nh.subscribe("debug/mode_shift", 10, &AutoDockingROS::debugCb, this);
+    // Publishers and subscribers
+    velocity_commander_ = this->create_publisher<geometry_msgs::msg::Twist>("commands/velocity", 10);
+    debug_jabber_ = this->create_publisher<std_msgs::msg::String>("debug/feedback", 10);
 
-  odom_sub_.reset(new message_filters::Subscriber<nav_msgs::Odometry>(nh, "odom", 10));
-  core_sub_.reset(new message_filters::Subscriber<kobuki_msgs::SensorState>(nh, "core", 10));
-  ir_sub_.reset(new message_filters::Subscriber<kobuki_msgs::DockInfraRed>(nh, "dock_ir", 10));
-  sync_.reset(new message_filters::Synchronizer<SyncPolicy>(SyncPolicy(10), *odom_sub_, *core_sub_, *ir_sub_));
-  sync_->registerCallback(boost::bind(&AutoDockingROS::syncCb, this, _1, _2, _3));
+    debug_ = this->create_subscription<std_msgs::msg::String>(
+      "debug/mode_shift", 10, std::bind(&AutoDockingROS::debugCb, this, _1)
+    );
 
-  return dock_.init();
-}
+    odom_sub_.reset(new message_filters::Subscriber<nav_msgs::msg::Odometry>(this, "odom"));
+    core_sub_.reset(new message_filters::Subscriber<kobuki_ros_interfaces::msg::SensorState>(this, "sensors/core"));
+    ir_sub_.reset(new message_filters::Subscriber<kobuki_ros_interfaces::msg::DockInfraRed>(this, "sensors/dock_ir"));
```

Comment:
`std::make_shared` is generally preferred to `reset(new)`; it handles exceptions better and is more performant.